src/tpm2-tss-engine-rand.c

```diff
@@ -67,12 +107,12 @@ rand_bytes(unsigned char *buf, int num)
         memcpy(buf, &b->buffer, b->size);
         num -= b->size;
         buf += b->size;
-        free(b);
     }
 
-    esys_ctx_free(&esys_ctx);
-
  end:
+    if(b)
+        free(b);
```

Comment:
if b is null free will ignore it (per the C stdlib spec), so just make sure it's initialized.

Reply:
yes